tests/MongoDB.EntityFrameworkCore.FunctionalTests/Query/CompositeKeyQueryTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Runtime.CompilerServices;
+using Microsoft.EntityFrameworkCore;
+
+namespace MongoDB.EntityFrameworkCore.FunctionalTests.Query;
+
+public class CompositeKeyQueryTests : IClassFixture<TemporaryDatabaseFixture>
+{
+    private readonly TemporaryDatabaseFixture _temporaryDatabase;
+
+    public CompositeKeyQueryTests(TemporaryDatabaseFixture fixture)
+    {
+        _temporaryDatabase = fixture;
+    }
+
+    [Fact]
+    public void Should_query_by_key_component()
+    {
+        var dbContext = CreateContext();
+
+        var result = dbContext.Entitites.Where(e => e.Key1 == "two").ToList();
+
+        Assert.Equal(2, result.Count);
+        Assert.All(result, e => Assert.Equal("two", e.Key1));
+    }
+
+    [Fact]
+    public void Should_query_by_key_components()
+    {
+        var dbContext = CreateContext();
+
+        var result = dbContext.Entitites.Where(e => e.Key1 == "two" && e.Key2 == 3).ToList();
+
+        Assert.Single(result);
+        Assert.All(result, e => Assert.Equal("two", e.Key1));
+        Assert.All(result, e => Assert.Equal(3, e.Key2));
+    }
```

Comment:
Do we support looking up key elements with non-equals operations?  e.g. e.Key2 > 3 etc.